bolt/lib/linear_solver/linear_solver.py

```diff
@@ -145,54 +195,59 @@ def _calculate_p(self):
                                                       p1_center,
                                                       p3_center)
 
+        # Flattening the obtained arrays:
         p1_center = af.flat(af.to_array(p1_center))
         p2_center = af.flat(af.to_array(p2_center))
         p3_center = af.flat(af.to_array(p3_center))
 
+        # Reordering such that variation in velocity is along axis 2:
+        # This is done to be consistent with the positionsExpanded form:
         p1_center = af.reorder(p1_center, 2, 3, 0, 1)
         p2_center = af.reorder(p2_center, 2, 3, 0, 1)
         p3_center = af.reorder(p3_center, 2, 3, 0, 1)
 
         af.eval(p1_center, p2_center, p3_center)
         return(p1_center, p2_center, p3_center)
 
-    # Assigning function that is used in computing the derivatives
-    # of the background distribution function:
-    _calculate_dfdp_background = calculate_dfdp_background
-
     def _initialize(self, params):
         """
         Called when the solver object is declared. This function is
-        used to initialize the mode perturbation of the distribution
-        function f_hat, along with the mode perturbation of the field
-        quantities
+        used to initialize the distribution function, and the field
+        quantities using the options as provided by the user. The
+        quantities are then mapped to the fourier basis by taking FFTs.
+        The independant modes are then evolved by using the linear
+        solver.
         """
-        f = af.broadcast(self.physical_system.initial_conditions.\
-                         initialize_f, self.q1_center, self.q2_center,
-                         self.p1, self.p2, self.p3, params
-                         )
-
+        f     = af.broadcast(self.physical_system.initial_conditions.\
+                             initialize_f, self.q1_center, self.q2_center,
+                             self.p1, self.p2, self.p3, params
+                             )
```

Comment:
What exactly does `af.broadcast` do in this context? Please add comments. I understand `initial_conditions(q1, q2, p1, p2, p3, params)` as taking in the arguments `q1, q2, p1, p2, p3, params` and returning some output. Why do we need `af.broadcast` and what does it do?